soroban-sdk-macros/src/derive_spec_fn.rs

```diff
@@ -152,7 +152,7 @@ pub fn derive_fn_spec(
                     MAX,
                 ),
             ));
-            VecM::<_, MAX>::default()
+            VecM::<_, { u32::MAX }>::default()
```

Comment:
I believe this is due to https://github.com/stellar/stellar-xdr/commit/0a621ec7811db000a60efae5b35f78dee3aa2533. There's no longer a limit of 10, so the logic above this needs to be cleaned up. @leighmcculloch is my understanding correct?